### PR TITLE
Updated Schema and removed hard coded region parameter for ASE

### PR DESCRIPTION
**Pull request by cocallaw**
- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Updated the Schema version of the ASS V2 to 2016-09-01
* Removed hard coded region for ASE and added supported regions as allowedValues to aseLocation
* Updated README.md with description and updated date in metadata.json

### CODE REVIEW

**Comment by cocallaw** [comment]
Updating to schema version 2016-09-01

https://github.com/Azure/azure-resource-manager-schemas/blob/91a87cb1aec8e0e6280cf9bcd63dc6002cdb257b/schemas/2016-09-01/Microsoft.Web.json

File: 201-web-app-asev2-create/azuredeploy.json [hunk 1]
```diff
@@ -10,9 +10,31 @@
         },
         "aseLocation": { 
             "type": "string",
-            "defaultValue": "South Central US",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline 201-web-app-asev2-create/azuredeploy.json]: remove allowedValues

File: 201-web-app-asev2-create/azuredeploy.parameters.json [hunk 2]
```diff
@@ -6,7 +6,7 @@
             "value": "GEN_UNIQUE"
         },
         "aseLocation": {
-            "value": "South Central US"
+            "value": "changeme"
```
Comment (bmoore-msft) [inline 201-web-app-asev2-create/azuredeploy.parameters.json]: see the readme for how to use samples with existing resources

File: 201-web-app-asev2-ilb-create/azuredeploy.json [hunk 3]
```diff
@@ -10,9 +10,31 @@
         },
         "aseLocation": { 
             "type": "string",
-            "defaultValue": "South Central US",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline 201-web-app-asev2-ilb-create/azuredeploy.json]: remove allowed values

File: 201-web-app-asev2-ilb-create/azuredeploy.json [hunk 4]
```diff
@@ -28,23 +50,27 @@
             }
         },
         "internalLoadBalancingMode": {
-            "type": "int",
+            "type": "string",
```
Comment (bmoore-msft) [inline 201-web-app-asev2-ilb-create/azuredeploy.json]: the type, default and allowed values will not work together

File: 201-web-app-asev2-ilb-create/azuredeploy.parameters.json [hunk 5]
```diff
@@ -15,7 +15,7 @@
             "value": "changeme"
         },
         "internalLoadBalancingMode": {
-            "value": 3
+            "value": "changeme"
```
Comment (bmoore-msft) [inline 201-web-app-asev2-ilb-create/azuredeploy.parameters.json]: remove this and use a valid default value in the template

File: 201-web-app-asev2-ilb-create/azuredeploy.parameters.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
             "value": "GEN-UNIQUE"
         },
         "aseLocation": {
-            "value": "South Central US"
+            "value": "changeme"
```
Comment (bmoore-msft) [inline 201-web-app-asev2-ilb-create/azuredeploy.parameters.json]: remove this, it's not valid

File: 201-web-app-asev2-create/azuredeploy.json (hunk 1 above)
Comment (cocallaw) [reply 201-web-app-asev2-create/azuredeploy.json]: Corrected

File: 201-web-app-asev2-create/azuredeploy.parameters.json (hunk 2 above)
Comment (cocallaw) [reply 201-web-app-asev2-create/azuredeploy.parameters.json]: Corrected

File: 201-web-app-asev2-ilb-create/azuredeploy.json (hunk 3 above)
Comment (cocallaw) [reply 201-web-app-asev2-ilb-create/azuredeploy.json]: Corrected

File: 201-web-app-asev2-ilb-create/azuredeploy.json (hunk 4 above)
Comment (cocallaw) [reply 201-web-app-asev2-ilb-create/azuredeploy.json]: Updated the default value to string

File: 201-web-app-asev2-ilb-create/azuredeploy.parameters.json (hunk 5 above)
Comment (cocallaw) [reply 201-web-app-asev2-ilb-create/azuredeploy.parameters.json]: Updated to a valid default value per schema

File: 201-web-app-asev2-ilb-create/azuredeploy.parameters.json (hunk 6 above)
Comment (cocallaw) [reply 201-web-app-asev2-ilb-create/azuredeploy.parameters.json]: Changed to valid value

**Comment by cocallaw** [comment]
All requested changes have been completed @bmoore-msft

**Pull request merged**